packages/salesforcedx-vscode-apex/src/oas/esrHandler.ts

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { WorkspaceContextUtil, workspaceUtils } from '@salesforce/salesforcedx-utils-vscode';
+import { RegistryAccess } from '@salesforce/source-deploy-retrieve-bundle';
+import { XMLBuilder, XMLParser } from 'fast-xml-parser';
+import * as fs from 'fs';
+import { OpenAPIV3 } from 'openapi-types';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { stringify } from 'yaml';
+import { workspaceContext } from '../context';
+import { nls } from '../messages';
+import GenerationInteractionLogger from '../oas/generationInteractionLogger';
+import { ApexOASInfo, ExternalServiceOperation } from '../oas/schemas';
+import { createProblemTabEntriesForOasDocument, getCurrentTimestamp, summarizeDiagnostics } from '../oasUtils';
+import { getTelemetryService } from '../telemetry/telemetry';
+import { ProcessorInputOutput } from './documentProcessorPipeline/processorStep';
+import { PromptGenerationOrchestrator } from './promptGenerationOrchestrator';
+
+/*
+ * Handles the creation and management of External Service Registration (ESR) metadata.
+ * This includes saving OpenAPI specifications as ESR metadata, managing named credentials,
+ * and building the ESR XML and YAML files.
+ */
+export class ESRHandler {
```

Comment:
Maybe change the name to ExternalServiceRegistrationDoc, or something like that.